dev/noapp/index.html

```diff
@@ -0,0 +1,171 @@
+            <div class="field-area">
+                <input type="text" id="token" autocomplete="on" required>
+                <label for="save_token">
+                    <span>Save Token in Local Storage?</span>
+                    <input type="checkbox" id="save_token" autocomplete="on">
+                </label>
+            </div>
+            <br>
+
+            <div class="submit-wrapper">
+                <button class="btn btn-default pull-left" type="button" id="token-management-btn" disabled title="Please enter an MDR url">
+                    <i class="fa fa-external-link"></i> Token Management
+                </button>
+                <button class="btn btn-primary" type="submit">Open Environment</button>
+            </div>
+        </form>
+    </div>
+    <script>
+        // format url to http://x.x.x
+        const formatUrl = (url) => {
+            url = url.trim()
+            if (!url.startsWith("http")) {
+                url = "http://" + url
+            }
+            url = url.replace(/\/$/, "") // remove trailing slash
+            return url
+        }
+        
+        const appInput = document.getElementById("app_url")
+        const mdrInput = document.getElementById("mdr_url")
+        const tokenInput = document.getElementById("token")
+        const saveTokenInput = document.getElementById("save_token")
+        
+        const tokenBtn = document.getElementById("token-management-btn")
+        const envForm = document.getElementById("environment-form")
+
+        // autofill from local storage
+        appInput.value = localStorage.getItem("app_url") || ""
+        mdrInput.value = localStorage.getItem("mdr_url") || ""
+        tokenInput.value = localStorage.getItem("token") || ""
+        saveTokenInput.checked = (localStorage.getItem("save_token") == "true") || false
+
+        // save to local storage on form submit (accessed by _top since /noapp is same origin)
+        envForm.addEventListener("submit", (event) => {
+            localStorage.setItem("app_url", event.target.app_url.value)
+            localStorage.setItem("save_token", event.target.save_token.checked)
+            localStorage.setItem("mdr_url", event.target.mdr_url.value)
+            
+            if (event.target.save_token.checked) {
+                localStorage.setItem("token", event.target.token.value)
+            } else {
+                localStorage.removeItem("token")
+
+                // don't redirect, send token to _top
+                event.preventDefault()
+
+                // update url with params
+                const search = window.location.search = new URLSearchParams({
+                    app_url: event.target.app_url.value,
+                    mdr_url: event.target.mdr_url.value}).toString()
+                window.history.pushState({}, window.title, `${window.location.href}?${search}`)
```

Comment:
this will cause a broken double query string. 

Starting URL: http://localhost:8080/noapp/

new URLSearchParams({...}).toString() creates: "app_url=foo&mdr_url=bar"
window.location.search = "app_url=foo&mdr_url=bar" causes browser to navigate to http://localhost:8080/noapp/?app_url=foo&mdr_url=bar (page reloads)

window.location.href is now http://localhost:8080/noapp/?app_url=foo&mdr_url=bar
pushState tries to set URL to http://localhost:8080/noapp/?app_url=foo&mdr_url=bar?app_url=foo&mdr_url=bar (broken double query string)